plugins/processors/ec2tagger/ec2tagger_test.go

```diff
@@ -275,26 +278,21 @@ func TestInitSuccessWithNoTagsVolumesUpdate(t *testing.T) {
 	assert.Equal(3, ec2Client.tagsCallCount)
 	assert.Equal(2, ec2Client.volumesCallCount)
 	//check tags and volumes
-	expectedTags := map[string]string{
-		"tagKey1":              "tagVal1",
-		"tagKey2":              "tagVal2",
-		"AutoScalingGroupName": "ASG-1",
-	}
+	expectedTags := map[string]string{tagKey1: tagVal1, tagKey2: tagVal2, "AutoScalingGroupName": tagVal3}
```

Comment:
Why is there inconsistency for the map keys? `tagKey1` is created as a var/const at the top of the file but `"AutoScalingGroupName"` gets redefined in all of the tests.

Reply:
This may explain why the inconsistency for ASG:
```
## Note: This plugin renames the "aws:autoscaling:groupName" EC2 Instance Tag key to be spelled "AutoScalingGroupName".
  ## This aligns it with the AutoScaling dimension-name seen in AWS CloudWatch.
  # ec2_instance_tag_keys = ["aws:autoscaling:groupName", "Name"]
```

Reply:
I don't understand what this means

Reply:
Based on my understanding, whenever CWAgent retrieves EC2 Instance Tags (not volume or not IMDS), the EC2 Instance will have the key tag as `aws:autoscaling:groupName` for AutoScalingGroup; then we will scrapped it; converted to `AutoScalingGroupName` as a key (back and forth) and appends the AutoScalingGroup as a dimension to the collected metrics (to match the `AutoScalingGroupName` key in CWAgent JSON configuration)

So long short story, the EC2 has tag `aws:autoscaling:groupName` (and its value) for AutoScalingGroup, we collected the `aws:autoscaling:groupName` tag, converted to `AutoScalingGroup` as a dimension. 

Let's me know if that make sense